cmd/lotus-shed/msg.go

```diff
@@ -36,6 +42,47 @@ var msgCmd = &cli.Command{
 			return err
 		}
 
+		api, closer, err := lcli.GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		ctx := lcli.ReqContext(cctx)
+
+		// Get the CID of the message
+		mcid := msg.Cid()
+
+		// Search for the message on-chain
+		lookup, err := api.StateSearchMsg(ctx, mcid)
+		if err != nil {
+			return err
+		}
+		if lookup == nil {
+			return fmt.Errorf("failed to find message: %s", mcid)
```

Comment:
This should work with messages which haven't yet landed on-chain - e.g. when you want to see a hex-encoded unsigned message before signing.